Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/RemoteXDA/RemoteXDAInstanceMain.tsx

```diff
@@ -22,161 +22,91 @@
-                <li className="nav-item" hidden={props.Roles.indexOf('Administrator') < 0} style={{ width: '15%', paddingRight: 10 }}>
-                    <fieldset className="border" style={{ padding: '10px', height: '100%' }}>
-                        <legend className="w-auto" style={{ fontSize: 'large' }}>Actions:</legend>
-                        <form>
-                            <button className="btn btn-primary" onClick={(event) => {
-                                if (props.Roles.indexOf('Administrator') > -1) {
-                                    event.preventDefault();
-                                    setShowNew(true);
-                                }
-                            }}>Add Remote Connection</button>
-                        </form>
-                    </fieldset>
-                </li>
-            </SearchBar>
-            <div style={{ width: '100%', height: 'calc( 100% - 136px)' }}>
-                <ReactTable.Table<OpenXDA.Types.RemoteXDAInstance>
-                    TableClass="table table-hover"
-                    Data={searchResults}
-                    SortKey={sortKey}
-                    Ascending={ascending}
-                    OnSort={(d) => {
-                        if (d.colKey === sortKey)
-                            setAscending(!ascending);
-                        else {
-                            setAscending(true);
-                            setSortKey(d.colField);
-                        }
-                    }}
-                    OnClick={handleSelect}
-                    TheadStyle={{ fontSize: 'smaller', display: 'table', tableLayout: 'fixed', width: '100%' }}
-                    TbodyStyle={{ display: 'block', overflowY: 'scroll', maxHeight: window.innerHeight - 300, width: '100%' }}
-                    RowStyle={{ fontSize: 'smaller', display: 'table', tableLayout: 'fixed', width: '100%' }}
-                    Selected={(item) => false}
-                    KeySelector={(item) => item.ID}
-                >
-                    <ReactTable.Column<OpenXDA.Types.RemoteXDAInstance>
-                        Key={'Name'}
-                        AllowSort={true}
-                        Field={'Name'}
-                        HeaderStyle={{ width: 'auto' }}
-                        RowStyle={{ width: 'auto' }}
-                    > Name
-                    </ReactTable.Column>
-                    <ReactTable.Column<OpenXDA.Types.RemoteXDAInstance>
-                        Key={'Address'}
-                        AllowSort={true}
-                        Field={'Address'}
-                        HeaderStyle={{ width: 'auto' }}
-                        RowStyle={{ width: 'auto' }}
-                    > URL
-                    </ReactTable.Column>
-                </ReactTable.Table>
-            </div>
-
-            <Modal Show={showNew} Title={'New Remote openXDA Instance Connection'}
+    return <>
+        <GenericByPage<OpenXDA.Types.RemoteXDAInstance>
```

Comment:
If you add and delete an instance, it doesn't go away until refresh.

Reply:
This bug wasn't totally replicable which is strange, but I made a change to make it more consistent with the ByPages which handle delete functions similarly. In most recent commit.

Reply:
More consistent in what way? Does this solve the issue?

Reply:
Consistent in what way?